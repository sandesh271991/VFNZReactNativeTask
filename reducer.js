export const GET_REPOS = 'my-awesome-app/repos/LOAD';
export const GET_REPOS_SUCCESS = 'my-awesome-app/repos/LOAD_SUCCESS';
export const GET_REPOS_FAIL = 'my-awesome-app/repos/LOAD_FAIL';


const initialState = {
  repos: [],
  loading: false,
  error: null
};


export default function reducer(state = initialState , action) {
  switch (action.type) {
    case GET_REPOS:
      return { ...state, loading: true };
    case GET_REPOS_SUCCESS:
      return { ...state, loading: false, repos: action.payload.data };
    case GET_REPOS_FAIL:
      return {
        ...state,
        loading: false,
        error: 'Error while fetching repositories',
      };
    default:
      return state;
  }
}

export function listRepos(photos) {
  return {
    type: GET_REPOS,
    payload: {
      request: {
        url: `photos/`
      }
    }
  };
}

export function listThumb(albumId) {
  return {
    type: GET_REPOS,
    payload: {
      request: {
        url: `photos?albumId=${albumId}`
      }
    }
  };
}