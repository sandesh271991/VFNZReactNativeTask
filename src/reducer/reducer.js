export const GET_ALBUM = 'album/LOAD';
export const GET_ALBUM_SUCCESS = 'album/LOAD_SUCCESS';
export const GET_ALBUM_FAIL = 'album/LOAD_FAIL';


const initialState = {
  albums: [],
  loading: false,
  error: null,
};


export default function reducer(state = initialState, action) {
  switch (action.type) {
    case GET_ALBUM:
      return { ...state, loading: true };
    case GET_ALBUM_SUCCESS:
      return { ...state, loading: false, albums: action.payload.data };
    case GET_ALBUM_FAIL:
      return {
        ...state,
        loading: false,
        error: 'Error while fetching albums',
      };
    default:
      return state;
  }
}

export function listRepos() {
  return {
    type: GET_ALBUM,
    payload: {
      request: {
        url: 'photos/',
      },
    },
  };
}

export function listThumb(albumId) {
  return {
    type: GET_ALBUM,
    payload: {
      request: {
        url: `photos?albumId=${albumId}`,
      },
    },
  };
}
