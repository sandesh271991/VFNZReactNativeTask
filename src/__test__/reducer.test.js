import expect from 'expect';
import reducer from '../reducer/reducer';

const initialState = {
  albums: [],
  loading: false,
  error: null,
};

/** Testing the state data in Reducer */
describe('post reducer', () => {
  it('should return the initial state', () => {
    expect(reducer(undefined, {})).toEqual(initialState);
  });
  it('is correct', () => {
    const action = { type: 'GET_ALBUM' };
    expect(reducer(undefined, action)).toMatchSnapshot();
  });
});
