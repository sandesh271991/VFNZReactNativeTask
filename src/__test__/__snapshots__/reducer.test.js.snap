// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`INITIAL_STATE is correct 1`] = `
Object {
  "albums": Array [],
  "error": null,
  "loading": false,
}
`;

exports[`post reducer is correct 1`] = `
Object {
  "albums": Array [],
  "error": null,
  "loading": false,
}
`;
